packages/consensus/src/types/utreexo.cairo

```diff
@@ -142,3 +226,267 @@ impl UtreexoBatchProofDisplay of Display<UtreexoBatchProof> {
         Result::Ok(())
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::{UtreexoState, UtreexoAccumulator, UtreexoProof, OutPoint};
+    use crate::types::transaction::{TxOut};
+
+
+    // Test the basic functionality of the Utreexo accumulator
+    ///
+    /// This test covers the following scenarios:
+    /// 1. Adding a single leaf and verifying it
+    /// 2. Adding a second leaf and verifying both leaves
+    /// 3. Adding a third leaf and verifying previous leaves
+    /// 4. Adding a fourth leaf and verifying all leaves
+    ///
+    /// For each scenario, the test:
+    /// - Initializes the UtreexoState with the appropriate roots and number of leaves
+    /// - Creates an OutPoint representing the leaf
+    /// - Constructs a UtreexoProof for the leaf
+    /// - Calls the verify function and asserts that it succeeds
+    ///
+    /// The test uses predefined txid values (0x111111..., 0x222222..., etc.) for simplicity.
+    /// It checks the correct root values at each stage of the Utreexo tree's growth.
+    #[test]
+    fn test_utreexo_basics() {
+        // Add the first leaf (0x111111111111111111111111)
+        let leaf1_txid = 0x111111111111111111111111;
+        let leaf1_txid_u256: u256 = leaf1_txid.into();
+
+        let leaf1 = OutPoint {
```

Comment:
You don't have to deal with outpoints in unit tests, the leaves are just felts

Reply:
but in the interface `verify` is currently accept an `OutPoint`

Reply:
I think having a helper to create mock outpoint is not a bad idea

Reply:
Right, let's change the verify interface to deal with felt252 instead (same thing is done for add)

Reply:
Still requires https://github.com/keep-starknet-strange/raito/pull/192/files#r1762930962